src/main/java/zone/rong/mixinbooter/MixinBooterMixinPlugin.java

```diff
@@ -0,0 +1,46 @@
+package zone.rong.mixinbooter;
+
+import net.minecraftforge.fml.common.Loader;
+import org.objectweb.asm.tree.ClassNode;
+import org.spongepowered.asm.mixin.extensibility.IMixinConfigPlugin;
+import org.spongepowered.asm.mixin.extensibility.IMixinInfo;
+
+import java.util.List;
+import java.util.Set;
+
+public class MixinBooterMixinPlugin implements IMixinConfigPlugin {
+
+    @Override
+    public void onLoad(String mixinPackage) {}
+
+    @Override
+    public String getRefMapperConfig() {
+        return null;
+    }
+
+    @Override
+    public boolean shouldApplyMixin(String targetClassName, String mixinClassName) {
+        String version = Loader.instance().getMCVersionString();
```

Comment:
`net.minecraftforge.fml.common.Loader` ?
Or, `(String)net.minecraftforge.fml.relauncher.data()[4]` ?

Reply:
> `net.minecraftforge.fml.common.Loader.instance().getMCVersionString()` ? I would recommend using `(String)net.minecraftforge.fml.relauncher.FMLInjectionData.data()[4]` .

https://github.com/MinecraftForge/FML/blob/d4ded9d6e218ac097990e836676bbe22b47e5966/src/main/java/net/minecraftforge/fml/relauncher/FMLInjectionData.java

The 1.8 version getProperty has chances to be missing, that can be caused by bad launcher installation

Reply:
> > `net.minecraftforge.fml.common.Loader.instance().getMCVersionString()` ? I would recommend using `(String)net.minecraftforge.fml.relauncher.FMLInjectionData.data()[4]` .
> 
> https://github.com/MinecraftForge/FML/blob/d4ded9d6e218ac097990e836676bbe22b47e5966/src/main/java/net/minecraftforge/fml/relauncher/FMLInjectionData.java
> 
> The 1.8 version getProperty has chances to be missing, that can be caused by bad launcher installation

Equal

https://github.com/MinecraftForge/FML/blob/d4ded9d6e218ac097990e836676bbe22b47e5966/src/main/java/net/minecraftforge/fml/common/launcher/FMLDeobfTweaker.java#L39

the data in `Loader` comes from `[FMLInjectionData](https://github.com/MinecraftForge/FML/blob/d4ded9d6e218ac097990e836676bbe22b47e5966/src/main/java/net/minecraftforge/fml/relauncher/FMLInjectionData.java)`

Reply:
There's no reason to get data from an Object[] that might contain unexpected content.
Just use the getter that can always return the thing wanted